src/components/multi-level-dropdown/multi-level-dropdown.js

```diff
@@ -0,0 +1,240 @@
+  };
+
+  getGroupedItems = () => {
+    const newItems = this.filterItems();
+    const groups = remove(Object.keys(newItems), r => r !== 'undefined');
+    const orphansList = newItems.undefined || [];
+    let listWithGroupsAndItems = orphansList;
+    groups.forEach(g => {
+      listWithGroupsAndItems = listWithGroupsAndItems.concat(newItems[g]);
+    });
+    return this.addIsActive(listWithGroupsAndItems);
+  };
+
+  addIsActive = itemList => {
+    const { values } = this.props;
+    const parentsWithActiveChilds = [];
+    const valueArray = castArray(values);
+    valueArray.forEach(v => {
+      if (v.group) parentsWithActiveChilds.push(String(v.group));
+    });
+    return itemList.map(i => ({
+      ...i,
+      active: valueArray.includes(i) ||
+        valueArray.map(v => v && v.value).includes(i.value),
+      hasActiveChild: parentsWithActiveChilds.includes(String(i.value))
+    }));
+  };
+
+  filterItems = () => {
+    const { inputValue } = this.state;
+    const { options, groupKey } = this.props;
+
+    return groupBy(
+      inputValue
+        ? options.filter(
+          item => deburrUpper(item.label).indexOf(deburrUpper(inputValue)) > -1
+        )
+        : options,
+      groupKey
+    );
+  };
+
+  handleMultiselectChange = (changes, downshiftStateAndHelpers) => {
+    const { values: selectedItems } = this.props;
+    const { onChange } = this.props;
+    const itemToRemove = selectedItems.find(
+      s => s.label === changes.inputValue
+    );
+    if (itemToRemove) {
+      selectedItems.splice(selectedItems.indexOf(itemToRemove), 1);
+    } else {
+      selectedItems.push(downshiftStateAndHelpers.selectedItem);
+    }
+    onChange(selectedItems);
+  };
+
+  updateActiveLabel = (selectedItem, clear) => {
+    const { values, multiselect } = this.props;
+    if (clear) return this.setState({ activeLabel: null });
+
+    const activeLabel = multiselect
+      ? values && values.length === 1 && values[0].label || null
+      : selectedItem;
+    return this.setState({ activeLabel });
+  };
+
+  handleStateChange = (changes, downshiftStateAndHelpers) => {
+    const { multiselect } = this.props;
+    if (changes) {
+      if (!downshiftStateAndHelpers.isOpen) {
+        this.setState({ inputValue: '' });
+      } else if (changes.type === EVENT_TYPES.mouseUp) {
+        this.setState({ isOpen: false });
+      } else if (changes.inputValue || changes.inputValue === '') {
+        if (
+          multiselect &&
+            (changes.type === EVENT_TYPES.keyDownEnter ||
+              changes.type === EVENT_TYPES.click)
+        ) {
+          this.handleMultiselectChange(changes, downshiftStateAndHelpers);
+        }
+        this.setState({ inputValue: changes.inputValue });
+      }
+
+      if (changes.selectedItem && !multiselect) {
+        this.setState({ isOpen: false, inputValue: '' });
+      }
+
+      if (Object.keys(changes).indexOf('isOpen') > -1) {
```

Comment:
Is this something like `isOpening`? If yes, maybe we can add `const isOpening = Object.keys(changes).indexOf('isOpen') > -1` will be faster to understand.